File: Backend/src/main/java/com/nexfin/backend/model/dto/response/TransactionResponse.java
package com.nexfin.backend.model.dto.response;

import com.nexfin.backend.model.entity.Transaction;
import java.math.BigDecimal;
import java.time.OffsetDateTime;
import java.util.UUID;

public record TransactionResponse(
        UUID transactionId,
        UUID sourceWalletId,
        UUID targetWalletId,
        BigDecimal amount,
        String type,
        String status,
        String reference,
        OffsetDateTime createdAt) {

    public static TransactionResponse from(Transaction transaction) {
        return new TransactionResponse(
                transaction.getId(),
                transaction.getSourceWallet().getId(),
                transaction.getTargetWallet().getId(),
                transaction.getAmount(),
                transaction.getType().name(),
                transaction.getStatus().name(),
                transaction.getReference(),
                transaction.getCreatedAt());
    }
}

File: Frontend/src/commonMain/kotlin/com/nexfin/frontend/network/models/response/TransactionResponse.kt
package com.nexfin.frontend.network.models.response

import kotlinx.serialization.Serializable

@Serializable
data class TransactionResponse(
    val transactionId: String,
    val sourceWalletId: String,
    val targetWalletId: String,
    val amount: Double,
    val type: String,
    val status: String,
    val reference: String,
    val createdAt: String
)
